modules/match_utils/__init__.py

```diff
@@ -4,14 +4,27 @@
 import tempfile
 import contextlib
 import subprocess
-from typing import Iterator
+from typing import List, Iterator, Optional
 from pathlib import Path
 from zipfile import ZipFile
 
 import controller_utils  # isort:skip
 
 
 class Match:
+    def __init__(self, argv: Optional[List[str]] = None) -> None:
+        args = self.parse_args(argv)
+        match_data = self.construct_match_data(args)
+
+        with self.temporary_arena_root(f'match-{match_data.match_number}'):
+            self.prepare_match(args.archives_dir, match_data)
+
+            self.run_match()
+
+            self.collate_logs(args.archives_dir, match_data)
+            self.archive_match_file(args.archives_dir, match_data)
+            self.archive_match_recordings(args.archives_dir)
```

Comment:
This is quite a surprising thing for an `__init__` function to do. I would strongly encourage not doing this.
In general it's not clear to me that having this as a class is useful - there's not really any use of it as an object here.